src/simulation/simulator.cpp

```diff
@@ -194,6 +194,21 @@ void Simulator::simulate() {
 	QSet<ItemBase *> itemBases;
 	QString spiceNetlist = m_mainWindow->getSpiceNetlist("Simulator Netlist", netList, itemBases);
 
+	//Select the type of analysis based on if there is an oscilloscope in the simulation
+	foreach (ItemBase * item, itemBases) {
+		if(item->family().toLower().contains("oscilloscope")) {
+			//TODO: Use TextUtils::convertFromPowerPrefixU function
+			double time_div = TextUtils::convertFromPowerPrefix(item->getProperty("time/div"), "s");
+			std::cout << "time/div: " << item->getProperty("time/div").toStdString() << " " << time_div << std::endl;
+			double maxSimTime = time_div * 10;
+			QString tranAnalysis = QString(".TRAN %1 %2").arg(maxSimTime/100).arg(maxSimTime);
+			spiceNetlist.replace(".OP", tranAnalysis);
+			//TODO: Handle several oscilloscopes
+			break;
+		}
+	}
+
+
```

Comment:
This probably answers my above question.

Reply:
Yes, as soon as there is an oscilloscope, the transitory analysis is used. Before the trans analysis only made sense with the oscilloscope as there was no animation. Since I added the animation, users could use the LEDs in the breadboard to see the effects on their circuits. 
So, yes, maybe it is a good idea to add the trans analysis as an option in the simulate button. However, in that case, how do we specify the simulation time? 
Could you implement the button? The last time I played with adding buttons took me ages to make it correctly.